src/Apps/Order/Routes/Shipping2/FulfillmentDetails.tsx

```diff
@@ -66,16 +74,150 @@ export const FulfillmentDetails: FC<FulfillmentDetailsProps> = props => {
+            addressValues.country
+          )
+
+          fulfillmentMutationValues = {
+            id: props.order.internalID,
+            fulfillmentType: requiresArtsyShipping
+              ? "SHIP_ARTA"
+              : FulfillmentType.SHIP,
+            phoneNumber,
+            shipping: { ...addressValues, phoneNumber: "" },
+          }
+
+          if (addressVerifiedBy) {
+            fulfillmentMutationValues.addressVerifiedBy = addressVerifiedBy
+          }
+        } else {
+          requiresArtsyShipping = false
+          fulfillmentMutationValues = {
+            id: props.order.internalID,
+            fulfillmentType: FulfillmentType.PICKUP,
+            phoneNumber: formValues.attributes.phoneNumber,
+            shipping: {
+              addressLine1: "",
+              addressLine2: "",
+              country: "",
+              name: "",
+              city: "",
+              postalCode: "",
+              region: "",
+              phoneNumber: "",
+            },
+          }
+        }
+
+        const result = await saveFulfillmentDetails.submitMutation({
+          variables: { input: fulfillmentMutationValues },
+        })
+
+        const orderOrError = result.commerceSetShipping?.orderOrError
+
+        if (orderOrError?.__typename === "CommerceOrderWithMutationFailure") {
+          shippingContext.helpers.handleExchangeError(
+            orderOrError.error,
+            logger
+          )
+          return
+        }
+
+        if (performUserAddressUpdates) {
+          await props.processUserAddressUpdates(formValues)
+        }
+
+        props.onFulfillmentDetailsSaved({ requiresArtsyShipping })
+      } catch (error) {
+        shippingContext.helpers.orderTracking.errorMessageViewed({
+          error_code: null,
+          title: "An error occurred",
+          message:
+            "Something went wrong. Please try again or contact orders@artsy.net.",
+          flow: "user selects a shipping option",
+        })
+
+        shippingContext.helpers.dialog.showErrorDialog()
+      }
+    },
+    [
+      saveFulfillmentDetails,
+      props,
+      shippingContext.parsedOrderData,
+      shippingContext.helpers,
+    ]
+  )
+
+  // Force-re-save fulfillment details with existing values to refresh shipping quotes
+  useEffect(() => {
+    const existingFulfillmentDetails =
+      shippingContext.parsedOrderData.savedFulfillmentDetails
+    if (
+      shippingContext.state.stage === "refresh_shipping_quotes" &&
+      existingFulfillmentDetails?.fulfillmentType === FulfillmentType.SHIP
+    ) {
+      submitFulfillmentDetails({
+        performUserAddressUpdates: false,
+        formValues: {
+          attributes: existingFulfillmentDetails.fulfillmentDetails as ShipValues["attributes"],
+          fulfillmentType: existingFulfillmentDetails?.fulfillmentType,
+        },
+      })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
```

Comment:
The actual implementation on the ticket. Perform a one-time address re-save if the page loads with a special new 'automatically refresh shipping quotes' stage.